logger.go

```diff
@@ -252,7 +253,13 @@ func (l *Logger) SetOutput(w io.Writer) {
 		isDiscard = 1
 	}
 	atomic.StoreUint32(&l.isDiscard, isDiscard)
-	l.re = lipgloss.NewRenderer(w)
+	// Reuse cached renderers
+	v, ok := registry.Load(w)
+	if !ok {
+		v = lipgloss.NewRenderer(w, termenv.WithColorCache(true))
+		registry.Store(w, v)
+	}
+	l.re = v.(*lipgloss.Renderer)
```

Comment:
💅 but could avoid one cast

```suggestion
	v, ok := registry.Load(w)
	if ok {
		l.re = v.(*lipgloss.Renderer)
	} else {
		v = lipgloss.NewRenderer(w, termenv.WithColorCache(true))
		registry.Store(w, v)
		l.re = v
	}
```